presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -400,6 +400,7 @@ nonReserved
     | START | TRANSACTION | COMMIT | ROLLBACK | WORK | ISOLATION | LEVEL
     | SERIALIZABLE | REPEATABLE | COMMITTED | UNCOMMITTED | READ | WRITE | ONLY
     | CALL
+    | SUBSTRING
```

Comment:
Please have a look on commit description.
Without this change there is a clash with token `SUBSTRING` (syntax `SUBSTRING(expression FROM x [FOR y])`). There is no fallback to `#functionCall` rule when `#substring` production fails.
When `SUBSTRING` token is added to `nonReserved` both productions could potentially match at the same level (`#functionCall` matches on the same level as `#substring`, with precise token match) so each can fall to another.
For `power` you didn't have such an issue because there is no high level parser token for `power` so there is no other production matching `power` but `#functionCall`.

The other possibility would be to add to `primaryExpression` `SUBSTRING '(' valueExpression ',' valueExpression (',' valueExpression)? ')'`  but it's not very elegant.
Other possibility would be to change `SUBSTRING` token into an inlined phrase, so it would be:
`'SUBSTRING(' valueExpression FROM valueExpression (FOR valueExpression)? ')'  #substring` instead of `SUBSTRING '(' valueExpression FROM valueExpression (FOR valueExpression)? ')'  #substring` but I think it's even worse.